pkg/startup/startup.go

```diff
@@ -720,6 +723,53 @@ func configureIPPools(ctx context.Context, client client.Interface) {
 		return
 	}
 
+	// If CIDRs weren't specified through the environment variables, check if they're present in kubeadm's
+	// config map.
+	if len(ipv4Pool) == 0 || len(ipv6Pool) == 0 {
+		clientset, err := createKubernetesClient()
+		if err == nil {
+			kubeadmConfig, err := clientset.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(KubeadmConfigConfigMap, metav1.GetOptions{})
+			if err != nil {
```

Comment:
I think we should handle two separate error cases:

- kubeadm configmap doesn't exist: this is OK and expected for a lot of environments, and shouldn't result in a warning log
- some other error: this is potentially an issue (e.g., failed to query the API server) and should result in a hard error, and probably should stop processing and exit. 

In the current iteration both of these cases are handled by a warning log.

Reply:
Fixed